api/observability/v1/clusterlogforwarder_types.go

```diff
@@ -170,8 +170,29 @@ type LimitSpec struct {
 	MaxRecordsPerSecond int64 `json:"maxRecordsPerSecond"`
 }
 
-// ConfigMapOrSecretKey encodes a reference to a single field in either a ConfigMap or Secret in the same namespace.
-type ConfigMapOrSecretKey struct {
+// ConfigReferenceFrom specifies the source used for the config reference.
+//
+// +kubebuilder:validation:Enum:=configMap;secret;
+type ConfigReferenceFrom string
+
+const (
+	// ConfigReferenceFromSecret specifies to use config from the spec'd secret
+	ConfigReferenceFromSecret ConfigReferenceFrom = "secret"
+
+	// ConfigReferenceFromConfigMap specifies to use config from the spec'd configMap
+	ConfigReferenceFromConfigMap ConfigReferenceFrom = "serviceAccountToken"
+)
+
+// ConfigReference encodes a configuration reference to a single field in either a ConfigMap or Secret in the same namespace.
+// +kubebuilder:validation:XValidation:rule="self.from == 'secret' && has(self.secret)", message="Additional secret spec is required when config reference is from a secret"
+// +kubebuilder:validation:XValidation:rule="self.from == 'configMap' && has(self.configMap)", message="Additional configMap spec is required when config reference is from a configMap"
+type ConfigReference struct {
+
+	// From is the source from where to find config
+	//
+	// +kubebuilder:validation:Required
+	From ConfigReferenceFrom `json:"from"`
```

Comment:
This will require "from" needed in every TLS spec in your changes below.  I am not a fan of it for that reason.   Seems unnecessary, but I'm sure there is a reason for such this type of composition.